ui/components/ui/form-field/form-field.js

```diff
@@ -277,4 +290,12 @@ FormField.propTypes = {
    * If used ensure the id prop is set on the input and a label element is present using htmlFor with the same id to ensure accessibility.
    */
   wrappingLabelProps: PropTypes.object,
+  /**
+   * Show max button inside the input field
+   */
+  maxButton: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+  /**
+   * Show colored value inside the input field
+   */
+  coloredValue: PropTypes.bool,
```

Comment:
These also seem context specific to the max cup spending component? `maxButton` could be renamed to something like `rightAccessory`. What does the `coloredValue` indicate? Is it always the same color? Should this be added as a conditional class to the formfield in the spending cup component?

Reply:
@georgewrmarshall You are right, I deleted context specific updates from `form-field` component and added in the spending cap component. [Here](https://github.com/MetaMask/metamask-extension/pull/15522/commits/a70f94956c880932ea53f17e9f54a7fe1641933f) are those changes that I made. 

It will be good to re-review this PR @georgewrmarshall, if you have time, because this PR is a blocker for this ticket: #15211.  😄 

Also @bschorchit comment [here](https://github.com/MetaMask/metamask-extension/issues/15367#issuecomment-1255392751) that this is a stretch goal to be merged to unblocked that ticket and to try to merge all this tickets by September 29th so to gets in the 10.20 release that will be out early in October.

I updated the `Error case` for custom spending cap component also in the latest commit @bschorchit. 😄